framework/test/unit/modules/nft/module.spec.ts

```diff
@@ -11,8 +11,744 @@
  *
  * Removal or modification of this copyright notice is prohibited.
  */
+
+import { utils } from '@liskhq/lisk-cryptography';
+import { codec } from '@liskhq/lisk-codec';
+import { BlockAssets } from '@liskhq/lisk-chain';
+import { NFTModule } from '../../../../src/modules/nft/module';
+import { createGenesisBlockContext } from '../../../../src/testing';
+import {
+	invalidSchemaEscrowSubstoreGenesisAssets,
+	invalidSchemaNFTSubstoreGenesisAssets,
+	invalidSchemaSupportedNFTsSubstoreGenesisAssets,
+	invalidSchemaUserSubstoreGenesisAssests,
+	validData,
+} from './init_genesis_state_fixtures';
+import { genesisNFTStoreSchema } from '../../../../src/modules/nft/schemas';
+import {
+	ALL_SUPPORTED_NFTS_KEY,
+	LENGTH_ADDRESS,
+	LENGTH_CHAIN_ID,
+	LENGTH_COLLECTION_ID,
+	LENGTH_NFT_ID,
+} from '../../../../src/modules/nft/constants';
+import { NFTStore } from '../../../../src/modules/nft/stores/nft';
+import { SupportedNFTsStore } from '../../../../src/modules/nft/stores/supported_nfts';
+import { UserStore } from '../../../../src/modules/nft/stores/user';
+import { EscrowStore } from '../../../../src/modules/nft/stores/escrow';
+
 describe('nft module', () => {
```

Comment:
I would go in more detail through the unit tests after there is more clarity about the problem I mention in the comment to the lines 302-316 of `module.ts`.

Reply:
Valid concern, it could be rationalized if certain cases are already validated (as in the loop over NFTs between lines 205 and 260;
- First check on line 213, ensures that an entry exists for the NFT in UserSubstore or EscrowSubstore.
- Second check on line  221, makes sure that the entry exists only in UserSubstore or EscrowSubstore.
- Third check on line 229, ensures that there are no duplicate entries for the NFT in UserSubstore.
- Fourth check on line 233, ensures that there are no duplicate entries for the NFT is EscrowSubstore.
- Fifth check on line 239, ensures that if owner is a chain there should be an entry in EscrowSubstore.

Before these checks,
The loop on line 194 and 203, makes sure;
- That the owner of the NFT is a valid address or chain.
- And NFT is not duplicated.

Once all this is validated, entries for NFT, User and Escrow store can be made, independently.

However, the check;
```if (genesisStore.supportedNFTsSubstore.length === 0)``` is invalid and have been removed.

Reply:
Resolved by latest commit.